test/test_ft_badchannel.m

```diff
@@ -0,0 +1,95 @@
+function test_ft_badchannel
+
+% WALLTIME 00:10:00
+% MEM 2gb
+
+dataset = dccnpath('/home/common/matlab/fieldtrip/datatrl/Subject01.ds');
+
+ft_debug off
+
+%%
+
+cfg = [];
+cfg.dataset = dataset;
+cfg.channel = 'MEG';
+cfg.trl = [1 7*3*300 0]; % first 7 trials, each of 3 seconds
+cfg.continuous = 'yes';
+cfg.demean = 'yes';
+datacnt = ft_preprocessing(cfg);
+
+cfg = [];
+cfg.length = 1;
+datatrl = ft_redefinetrial(cfg, datacnt);
+
+%%
+
+cfg = [];
+cfg.method = 'template';
+neighbours = ft_prepare_neighbours(cfg, datatrl);
+
+%%
+
+% this can be used to determine the thresholds
+cfg = [];
+cfg.method = 'summary';
+cfg.neighbours = neighbours;
+data_clean = ft_rejectvisual(cfg, datatrl);
+
+%%
+% each method has its own optimal threshold
+% FT_REJECTVISUAL can be used to determine the appropriate value
+
+
+
+cfg = [];
+cfg.feedback = 'yes';
+cfg.neighbours = neighbours;
+
+% cfg.method = 'range';
```

Comment:
in principle, this can all be commented in, right (+ followed by a call to ft_badchannel), so that it just cycles through those possibilities. On the other hand perhaps there is not much added value in doing so.

Reply:
the test script is indeed not optimal yet, because it does not go through all options. It is more for manual testing at the moment. This will be updated later!